### PR TITLE
feat(compress): max compression w/ brotli & gzip

### PR DESCRIPTION
**Pull request by jbmoelker**
## Summary

Recipe including:

* npm task
* nginx config
* docs

## Checklist

* [x] The changes only affect or contain one recipe.
* [x] The changes are according to [contribution guidelines](/CONTRIBUTING.md#guidelines).
* [x] The new recipe has an entry in the [recipes index](/README.md#recipes) (only in case of new recipe).
* [x] The changes can be merged into the target branch without conflicts.

### CODE REVIEW

File: compress-with-brotli-and-gzip/scripts/compress.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+const gzipCompress = () => fileStream()
+    .pipe(gzip(gzipSettings))
+    .pipe(gulp.dest(outputDir));
+
+const compress = () => merge([
+		gzipCompress(),
+		brotliCompress()
+	])
+    .on('end', () => console.log(`Compressed files saved to ${outputDir}`));
```
Comment (thadeetrompetter) [inline compress-with-brotli-and-gzip/scripts/compress.js]: This task can really start taking its sweet time if you have a lot of files to compress. May i suggest introducing a [spinner](https://www.npmjs.com/package/ora) here to give those running the task some feedback that stuff is happening?

File: compress-with-brotli-and-gzip/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+
+### Configure server to use precompressed files
+
+You'll need to configure your server to serve the precompressed files when available.
+In addition you can enable on-the-fly compression of dynamic assets.
+
+* [Nginx configuration for serving precompressed files](lib/static-compression-nginx.conf)
+
+
```
Comment (thadeetrompetter) [inline compress-with-brotli-and-gzip/README.md:47]: it's mentioned in the blog article, but maybe include a [link to Brotli Nginx module](https://github.com/google/ngx_brotli) here to make it perfectly clear that action is required to get Brotli to work with nginx?

File: compress-with-brotli-and-gzip/lib/static-compression-nginx.conf [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Include this configuration inside http {} block:
```
Comment (thadeetrompetter) [inline compress-with-brotli-and-gzip/lib/static-compression-nginx.conf]: It's also OK to include in **server** or **location**

File: compress-with-brotli-and-gzip/scripts/compress.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+const brotliCompress = () => fileStream()
+    .pipe(brotli.compress(brotliSettings))
+    .pipe(gulp.dest(outputDir));
+
+const gzipCompress = () => fileStream()
+    .pipe(gzip(gzipSettings))
+    .pipe(gulp.dest(outputDir));
+
+const compress = () => merge([
+		gzipCompress(),
```
Comment (thadeetrompetter) [inline compress-with-brotli-and-gzip/scripts/compress.js]: mixed tabs & spaces here.

**Comment by jbmoelker** [comment]
@thadeetrompetter I agree with all your feedback. Latest commit contains:
* adjusted conf instructions w/ link to Brotli module
* Ora spinner
* Whitespace fix

**Pull request merged**